src/include/miopen/graphapi/matmul.hpp

```diff
@@ -82,5 +85,113 @@ class BackendMatmulDescriptor : public BackendDescriptor
     Matmul* getMatmul() noexcept { return &mMatmul; }
 };
 
+class OperationMatmul : public OpNode
+{
+private:
+    Tensor* mA;
+    Tensor* mB;
+    Tensor* mC;
+    int64_t mBatchCount = 1;
+    Tensor* mGemmMOverride;
+    Tensor* mGemmNOverride;
+    Tensor* mGemmKOverride;
+    Matmul* mMatmul;
+
+public:
+    OperationMatmul(Tensor* A,
+                    Tensor* B,
+                    Tensor* C,
+                    int batchCount,
+                    Tensor* MOverride,
+                    Tensor* NOverride,
+                    Tensor* KOverride,
+                    Matmul* matmul) noexcept
+        : mA(A),
+          mB(B),
+          mC(C),
+          mBatchCount(batchCount),
+          mGemmMOverride(MOverride),
+          mGemmNOverride(NOverride),
+          mGemmKOverride(KOverride),
+          mMatmul(matmul)
+    {
+    }
+
+    OperationMatmul() = default;
+    Tensor* getA() const { return mA; }
+    Tensor* getB() const { return mB; }
+    Tensor* getC() const { return mC; }
+    int64_t getBatchCount() const { return mBatchCount; }
+    Tensor* getMOverride() { return mGemmMOverride; }
+    Tensor* getNOverride() { return mGemmNOverride; }
+    Tensor* getKOverride() { return mGemmKOverride; }
+    Matmul* getMatmul() { return mMatmul; }
+    virtual std::vector<Tensor*> getInTensors() const override { return {getA(), getB()}; }
+    virtual std::vector<Tensor*> getOutTensors() const override { return {getC()}; }
+    virtual const std::string& signName() const override
+    {
+        static const std::string name = "OP_MATMUL";
+        return name;
+    }
+
+private:
+    friend class OperationMatmulBuilder;
+};
+
+class OperationMatmulBuilder
+{
+private:
+    OperationMatmul mOperationMatmul;
+    bool aSet      = false;
+    bool bSet      = false;
+    bool cSet      = false;
+    bool matmulSet = false;
```

Comment:
```suggestion
    bool mASet      = false;
    bool mBSet      = false;
    bool mCSet      = false;
    bool mMatmulSet = false;
```
Colleagues insist on sticking to our new naming rules.

Reply:
fixed